distribution/sender/routes/database-event-reader.xml

```diff
@@ -0,0 +1,115 @@
+
+                    <setProperty name="event">
+                        <simple>${body[0].event}</simple>
+                    </setProperty>
+	                 
+                 	<doTry>
+                 		<setProperty name="is-operation-completed">
+				            <spel>false</spel>
+				        </setProperty>
+				        
+	                    <log message="Loaded event from staging table: ${exchangeProperty.event}" loggingLevel="INFO" />
+	                     
+                        <setProperty name="route-retry-count-map">
+	                    	<spel>#{new java.util.HashMap()}</spel>
+	                	</setProperty>
+	                	
+		                <split>
+		                    <simple>{{db-event.destinations}}</simple>
+		                    <script>
+		                        <spel>#{getProperty('route-retry-count-map').put(body.trim(), 0)}</spel>
+		                    </script>
+		                </split>
+		                
+		                <setProperty name="table-names">
+		                    <!-- We need to look up failed events for rows in both the parent and subclass tables -->
+		                    <method beanType="org.openmrs.eip.component.utils.Utils" method="getTablesInHierarchy(${exchangeProperty.event.tableName})" />
+		                </setProperty>
+		
+		                <toD uri="jpa:SenderRetryQueueItem?query=SELECT r from SenderRetryQueueItem r WHERE r.event.tableName IN (${exchangeProperty.table-names}) AND r.event.primaryKeyId='${exchangeProperty.event.primaryKeyId}'" />
+		
+		                <split stopOnException="true">
+		                    <simple>${body}</simple>
+		                    <choice>
+		                        <when>
+		                            <simple>${exchangeProperty.route-retry-count-map.containsKey(${body.route.trim()})} != true</simple>
+		                            <throwException exceptionType="java.lang.Exception" message="No listener route found with name ${body.route.trim()}" />
+		                        </when>
+		                        <otherwise>
+		                            <script>
+		                                <spel>#{getProperty('route-retry-count-map').put(body.route, getProperty('route-retry-count-map').get(body.route.trim()) + 1)}</spel>
+		                            </script>
+		                        </otherwise>
+		                    </choice>
+		                </split>
+		                
+		                <log message="Route and retryCount map -> ${exchangeProperty.route-retry-count-map}" loggingLevel="INFO" />
+	                     
+						<setBody>
+			            	<simple>${exchangeProperty.event}</simple>
+			          	</setBody>
+			
+		               	<to uri="direct:db-event-processor" />
+		                
+						<setProperty name="is-operation-completed">
+				            <spel>true</spel>
+				        </setProperty>
+		                
+	                    <doCatch>
```

Comment:
Remove the try catch finally tags, let the error bubble out and handle it in the errorHandler set on the route

Reply:
Ok